runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowMetrics.java

```diff
@@ -280,22 +296,25 @@ private boolean isMetricTentative(MetricUpdate metricUpdate) {
      * @return a {@link MetricKey} that can be hashed and used to identify a metric.
      */
     private @Nullable MetricKey getMetricHashKey(MetricUpdate metricUpdate) {
-      String internalStepName = metricUpdate.getName().getContext().get("step");
-      String userStepName = getUserStepName(internalStepName);
-
-      if (userStepName == null
-          && (dataflowPipelineJob.transformStepNames == null
-              || !dataflowPipelineJob.transformStepNames.inverse().containsKey(internalStepName))) {
-        // If we can't translate internal step names to user step names, we just skip them
-        // altogether.
+      // If we can't translate internal step names to user step names, give up early
+      if (dataflowPipelineJob.transformStepNames == null) {
```

Comment:
We do not need "dataflowPipelineJob.transformStepNames" for portable job submission. So we should not return here if we can find a valid step name.

Reply:
Good catch. I duplicated this check in merge resolution.